### PR TITLE
feat(config): add bloom filter config

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#5176 

## What's changed and what's your intention?

Add config for bloom filter.

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

File: config/datanode.example.toml [hunk 1]
```diff
@@ -576,6 +576,39 @@ apply_on_query = "auto"
...
+mem_threshold_on_create = "auto"
+
+## Cache size for bloom filter metadata.
+metadata_cache_size = "4MiB"
+
+## Cache size for bloom filter content.
+content_cache_size = "128MiB"
+
+## Page size for bloom filter content cache.
+content_cache_page_size = "8MiB"
```
Comment (evenyag) [inline config/datanode.example.toml:610]: non-blocking: Configuring cache size for each index type is verbose. We may need to provide a global index cache.

File: src/datatypes/src/schema.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ use snafu::{ensure, ResultExt};
 use crate::error::{self, DuplicateColumnSnafu, Error, ProjectArrowSchemaSnafu, Result};
 use crate::prelude::ConcreteDataType;
 pub use crate::schema::column_schema::{
-    ColumnSchema, FulltextAnalyzer, FulltextOptions, Metadata, SkippingIndexOptions,
+    ColumnSchema, FulltextAnalyzer, FulltextOptions, Metadata, SkipIndexType, SkippingIndexOptions,
```
Comment (evenyag) [inline src/datatypes/src/schema.rs:31]: This is outdated, we may need to rebase https://github.com/GreptimeTeam/greptimedb/pull/5236

File: src/mito2/src/config.rs [hunk 3]
```diff
@@ -511,6 +514,58 @@ impl FulltextIndexConfig {
...
+    pub apply_on_query: Mode,
+    /// Memory threshold for creating the index.
+    pub mem_threshold_on_create: MemoryThreshold,
+
+    /// Cache size for metadata of the index. Setting it to 0 to disable the cache.
+    pub metadata_cache_size: ReadableSize,
+    /// Cache size for the index content. Setting it to 0 to disable the cache.
+    pub content_cache_size: ReadableSize,
+    /// Page size for the index content.
+    pub content_cache_page_size: ReadableSize,
```
Comment (evenyag) [inline src/mito2/src/config.rs:536]: We need to find a better way to config the index cache.

**Review by evenyag — APPROVED** [review]
Generally LGTM.

File: config/datanode.example.toml
Comment (waynexia) [inline config/datanode.example.toml:1]: sugg: change to "bloom_filter_index" for consistancy

**Review by waynexia — APPROVED** [review]

**Pull request merged**